Base/QTCore/qSlicerCoreIOManager.cxx

```diff
@@ -483,6 +483,9 @@ QString qSlicerCoreIOManager::forceFileNameValidCharacters(const QString& filena
   // Remove leading and trailing spaces
   sanitizedFilename = sanitizedFilename.trimmed();
 
+  // Ensure the filename doesn't exceed the maximum allowed length
+  sanitizedFilename = QString::fromStdString(vtkMRMLScene::ClampFileNameLength(sanitizedFilename.toStdString()));
```

Comment:
add to the method documentation that valid characters include the total number of characters (so long filenames will be shortened)